src/biocommons/seqrepo/dataproxy.py

```diff
@@ -0,0 +1,297 @@
+        aliases = list(set(md["aliases"]))  # ensure uniqueness
+        if namespace is not None:
+            nsd = namespace + ":"
+            aliases = [a for a in aliases if a.startswith(nsd)]
+        return aliases
+
+
+class _SeqRepoDataProxyBase(_DataProxy):
+    # wraps seqreqpo classes in order to provide translation to/from
+    # `ga4gh` identifiers.
+
+    def get_metadata(self, identifier):
+        md = self._get_metadata(identifier)
+        md["aliases"] = list(a for a in md["aliases"])
+        return md
+
+    def get_sequence(self, identifier, start=None, end=None):
+        return self._get_sequence(identifier, start=start, end=end)
+
+    @abstractmethod
+    def _get_metadata(self, identifier):  # pragma: no cover
+        pass
+
+    @abstractmethod
+    def _get_sequence(self, identifier, start=None, end=None):  # pragma: no cover
+        pass
+
+
+class SeqRepoDataProxy(_SeqRepoDataProxyBase):
+    """DataProxy based on a local instance of SeqRepo"""
+
+    def __init__(self, sr):
+        super().__init__()
+        self.sr = sr
+
+    def _get_sequence(self, identifier, start=None, end=None):
+        # fetch raises KeyError if not found
+        return self.sr.fetch_uri(identifier, start, end)
+
+    def _get_metadata(self, identifier):
+        ns, a = coerce_namespace(identifier).split(":", 2)
+        r = list(self.sr.aliases.find_aliases(namespace=ns, alias=a))
+        if len(r) == 0:
+            raise KeyError(identifier)
+        seq_id = r[0]["seq_id"]
+        seqinfo = self.sr.sequences.fetch_seqinfo(seq_id)
+        aliases = self.sr.aliases.find_aliases(seq_id=seq_id)
+        md = {
+            "length": seqinfo["len"],
+            "alphabet": seqinfo["alpha"],
+            "added": _isoformat(seqinfo["added"]),
+            "aliases": [f"{a['namespace']}:{a['alias']}" for a in aliases],
+            }
+        return md
+
+
+class SeqRepoRESTDataProxy(_SeqRepoDataProxyBase):
+    """DataProxy based on a REST instance of SeqRepo, as provided by seqrepo-rest-services"""
+
+    rest_version = "1"
+
+    def __init__(self, base_url):
+        super().__init__()
+        self.base_url = f"{base_url}/{self.rest_version}/"
+
+    def _get_sequence(self, identifier, start=None, end=None):
+        url = self.base_url + f"sequence/{identifier}"
+        _logger.info("Fetching %s", url)
+        params = {"start": start, "end": end}
+        resp = requests.get(url, params=params)
+        if resp.status_code == 404:
+            raise KeyError(identifier)
+        resp.raise_for_status()
+        return resp.text
+
+    def _get_metadata(self, identifier):
+        url = self.base_url + f"metadata/{identifier}"
+        _logger.info("Fetching %s", url)
+        resp = requests.get(url)
+        if resp.status_code == 404:
+            raise KeyError(identifier)
+        resp.raise_for_status()
+        data = resp.json()
+        return data
+
+
+class SequenceProxy(Sequence):
```

Comment:
is this class needed?

Reply:
i didn't remove this in my PR to your fork because i wasn't sure if you had time to review this ☝️